test_runner/regress/test_neon_superuser.py

```diff
@@ -76,3 +76,7 @@ def check_that_changes_propagated():
             assert [r[0] for r in res] == [10, 20, 30, 40]
 
         wait_until(10, 0.5, check_that_changes_propagated)
+
+        # Test that pg_monitor is working
+        cur.execute("SELECT query from pg_stat_activity LIMIT 1")
+        assert cur.fetchall()[0][0] != "<insufficient privilege>"
```

Comment:
```suggestion
        # Test that pg_monitor is working for neon_superuser role
        cur.execute("SELECT query from pg_stat_activity LIMIT 1")
        assert cur.fetchall()[0][0] != "<insufficient privilege>"
        # Test that pg_monitor is not working for non neon_superuser role without grant
        cur.execute("CREATE ROLE not_a_superuser LOGIN PASSWORD 'Password42!'")
        cur.execute("GRANT not_a_superuser TO neon_superuser WITH ADMIN OPTION")
        cur.execute("SET ROLE not_a_superuser")
        cur.execute("SELECT query from pg_stat_activity LIMIT 1")
        assert cur.fetchall()[0][1] == "<insufficient privilege>"
        cur.execute("RESET ROLE")
        # Test that pg_monitor is working for non neon_superuser role with grant
        cur.execute("GRANT pg_monitor TO not_a_superuser")
        cur.execute("SET ROLE not_a_superuser")
        cur.execute("SELECT query from pg_stat_activity LIMIT 1")
        assert cur.fetchall()[0][0] != "<insufficient privilege>"
        cur.execute("RESET ROLE")
        cur.execute("DROP ROLE not_a_superuser")
        # test that pg_monitor also allows us to read from pg_stat_progress_vacuum
        # pg_stat_progress_vacuum
        tableddl = """CREATE TABLE pgbench_accounts (
                        aid bigint NOT NULL,
                        bid integer,
                        abalance integer,
                        filler character(84),
                        -- create more blocks more quickly to make sure auto-vacuum has something to do
                        text_column_plain TEXT  DEFAULT repeat('NeonIsCool', 5),
                        jsonb_column_extended JSONB  DEFAULT ('{ "tell everyone": [' || repeat('{"Neon": "IsCool"},',9) || ' {"Neon": "IsCool"}]}')::jsonb
                    )
                    WITH (fillfactor='100')"""
        cur.execute(tableddl)
        cur.execute("insert into pgbench_accounts(aid,bid,abalance,filler) select aid, (aid - 1) / 100000 + 1, 0, '' from generate_series(1, 20000000) as aid")
        # wait until auto-vacuum starts
        def check_that_vacuum_is_running():
            cur.execute("SELECT count(*) FROM pg_stat_progress_vacuum")
            assert cur.fetchall()[0][0] != 0

        wait_until(10, 10.0, check_that_vacuum_is_running)
        # Test that pg_monitor is working for neon_superuser role and it can read phase column
        cur.execute("SELECT count(*) from pg_stat_progress_vacuum where phase is not NULL")
        assert cur.fetchall()[0][0] != 0 
```

Reply:
I can currently not test my proposal because of https://neondb.slack.com/archives/C033RQ5SPDH/p1707473061742059

Reply:
I had to remove the vacuum test because it was taking too long